src/services/cache-config/CacheConfigService.ts

```diff
@@ -315,10 +315,13 @@ export class CacheConfigService {
         this.createHelper(builder, address.concat("indexing", "INDEXING"), config.indexing);
         this.createHelper(builder, address.concat("compatibility", "COMPATIBILITY"), config.compatibility);
         this.createHelper(builder, address.concat("partition-handling", "PARTITION_HANDLING"), config["partition-handling"]);
-        this.createHelper(builder, address.concat("transaction", "TRANSACTION"), config.transaction);
+        if (this.isValidTXMode(builder, address, config)) {
+          this.createHelper(builder, address.concat("transaction", "TRANSACTION"), config.transaction);
+        } else {
+          deferred.reject("Cannot set TX mode to FULL_XA with ASYNC cache mode");
+        }
```

Comment:
@ryanemerson right, we do even when we update. Let me look at the createCacheStore method